src/TokenKind.php

```diff
@@ -10,205 +10,308 @@ class TokenKind {
+    const HeredocStart = 286;
+    const BacktickToken = 287;
+
+    // end_expression
+
+
+    const ScriptSectionStartTag = 317;
+
+    // begin_statement
+    const ScriptSectionEndTag = 318;
+
+    // end_statement
+    const ScriptSectionStartWithEchoTag = 319;
 
     // TODO how to handle incremental parsing w/ this?
-    const ScriptSectionPrependedText = 325;
-    const VoidReservedWord = 326;
-    const FalseReservedWord = 327;
-
-    const MemberName = 328;
-    const Expression = 329;
-
-    const BinaryReservedWord = 330; // TODO better way
-    const BooleanReservedWord = 331;
-    const DoubleReservedWord = 332;
-    const IntegerReservedWord = 333;
-    const ObjectReservedWord = 334;
-    const RealReservedWord = 335;
-    const ReturnType = 336;
-    const InlineHtml = 337;
-    const PropertyType = 338;
-
-//    const DollarOpenCurly = 339;
+    const ScriptSectionPrependedText = 320;
+    const MemberName = 321;
+    const Expression = 322;
+    const ReturnType = 323;
+    const InlineHtml = 324;
+    const PropertyType = 325;
+
+    // DollarOpenCurly
+
     const EncapsedAndWhitespace = 400;
-    const SingleQuoteToken = 401;
-    const DoubleQuoteToken = 402;
-    const DollarOpenBraceToken = 403;
-    const OpenBraceDollarToken = 404;
-    const CastToken = 405;
-    const HeredocStart = 406;
-    const HeredocEnd = 407;
-    const StringVarname = 408;
-    const UnsetCastToken = 409;
-    const StringCastToken = 410;
-    const ObjectCastToken = 411;
-    const IntCastToken = 412;
-    const DoubleCastToken = 413;
-    const BoolCastToken = 414;
-    const ArrayCastToken = 415;
-    const IntegerLiteralToken = 416;
-    const CommentToken = 417;
-    const DocCommentToken = 418;
+    const DollarOpenBraceToken = 401;
+    const OpenBraceDollarToken = 402;
+    const CastToken = 403;
+    const HeredocEnd = 404;
+    const StringVarname = 405;
+
+    // begin_cast
+    const UnsetCastToken = 406;
+    const StringCastToken = 407;
+    const ObjectCastToken = 408;
+    const IntCastToken = 409;
+    const DoubleCastToken = 410;
+    const BoolCastToken = 411;
+    const ArrayCastToken = 412;
+
+    // end_cast
+    const IntegerLiteralToken = 413;
+    const CommentToken = 414;
+    const DocCommentToken = 415;
 
     // TODO type annotations - PHP7
+
+    public static function isExpression(int $tokenKind): bool
+    {
+        return (2 === $tokenKind) || (147 < $tokenKind && $tokenKind < 201) ||(263 < $tokenKind && $tokenKind < 317);
+    }
+
+    public static function isNameOrKeywordOrReserved(int $tokenKind): bool
+    {
+        return 5 < $tokenKind && $tokenKind < 215;
+    }
+
+    public static function isKeyword(int $tokenKind): bool
+    {
+        return 6 < $tokenKind && $tokenKind < 201;
+    }
+
+    public static function isParameterTypeDeclaration(int $tokenKind): bool
+    {
+        return (101 < $tokenKind && $tokenKind < 104) || (203 < $tokenKind && $tokenKind < 209);
```

Comment:
The constants for token kinds vary between PHP versions. What works in PHP 7.0 likely won't work in PHP 8.0 and not in PHP 7.4. There's also no guarantee that php-src wouldn't reorder token kinds.